Source/Dafny/Compilers/Compiler-java.cs

```diff
@@ -2207,8 +2221,12 @@ private class GenericArrayElementLvalue : ILvalue {
           return false;
         }
       }
+
+      var targetDirectory = Path.GetDirectoryName(targetFilename);
+      EmitRuntimeJar(targetDirectory);
```

Comment:
Hmm...this should probably only happen if `DafnyOptions.O.UseRuntimeLib` is `false`. If it's `true`, the assumption is that you'll add `DafnyRuntime.jar` to the classpath yourself when you compile/run later. How hard is it to extract `DafnyRuntime.jar` from the nuget package after you've installed it?

I don't think this is blocking since existing codebases since we're still publishing the zip files as well, but we should probably make sure we start publishing `DafnyRuntime.jar` as well before we STOP publishing the zip files.

Reply:
That's an easy check to add, so I'll do it even if it's not immediately critical.

Reply:
Oh, and in response to your question, I don't know of any way to extract the embedded files from an assembly other than through the .NET API, but there probably is one.